src/yupSchema.js

```diff
@@ -17,9 +17,16 @@ export const baseString = string().trim();
 const todayDateHint = new Date().toISOString().substring(0, 10);
 const convertBytesToKB = convertDiskSizeFromTo('Bytes', 'KB');
 const convertKbBytesToMB = convertDiskSizeFromTo('KB', 'MB');
+const useOnlyStrings = (value, originalValue) => {
+  if (typeof originalValue === 'string') {
+    return originalValue;
+  }
+
+  return null;
+};
 
 const yupSchemas = {
-  text: string().trim().nullable(),
+  text: string().transform(useOnlyStrings).trim().nullable(),
```

Comment:
The reason why I use the transform was to bypass the originalValues to avoid the coercion that the string constructor [does](https://github.com/jquense/yup/blob/v0.29.1/src/string.js#L21).

I added the expect to have bigger confidence.

I'm trying to investigate how to do this with the .test and be able to cover all cases

Reply:
I found something in the yup documentation related to the coercion.

It seems if we pass the strict option as true, it doesn't coerce the values and it's able to run things right

Reply:
If you need help to figure out how to use test, let me know!

Reply:
To be able to use .test efficiently and access the context parameter, we have updated the yup library to 0.30.0.

See this [diff](https://github.com/remoteoss/json-schema-form/pull/12/files#diff-6a53a32c11d712a7acaaca4cf74985f7a76698a646b77c89d6a8841a67f61bacR24)

Reply:
Did the last changes and now tests are not breaking anymore